pkg/adaptor/cloud/azure/provider.go

```diff
@@ -179,46 +179,101 @@ func (p *azureProvider) CreateInstance(ctx context.Context, podName, sandboxID s
 		return nil, err
 	}
 
-	vmParameters := armcompute.VirtualMachine{
-		Location: to.Ptr(p.serviceConfig.Region),
-		Properties: &armcompute.VirtualMachineProperties{
-			HardwareProfile: &armcompute.HardwareProfile{
-				VMSize: to.Ptr(armcompute.VirtualMachineSizeTypes(p.serviceConfig.Size)),
-			},
-			StorageProfile: &armcompute.StorageProfile{
-				ImageReference: &armcompute.ImageReference{
-					ID: to.Ptr(p.serviceConfig.ImageId),
+	var vmParameters armcompute.VirtualMachine
+	if !p.serviceConfig.DisableCVM {
```

Comment:
I see OSDisk and SecurityProfile as the difference in armcompute/.VirtualMachine when using CVM. Can we just use these attributes if it's a CVM ?

Reply:
Make sense. Updated